src/rdata.cpp

```diff
@@ -41,51 +42,53 @@ ReturnData::ReturnData(std::vector<realtype> ts, int np, int nk, int nx,
 
     switch (rdata_reporting) {
     case RDataReporting::full:
-        initializeFullReporting();
+        initializeFullReporting(quadratic_llh);
         break;
 
     case RDataReporting::residuals:
-        initializeResidualReporting();
+        initializeResidualReporting(quadratic_llh);
         break;
 
     case RDataReporting::likelihood:
-        initializeLikelihoodReporting();
+        initializeLikelihoodReporting(quadratic_llh);
         break;
     }
 }
 
-void ReturnData::initializeLikelihoodReporting() {
+void ReturnData::initializeLikelihoodReporting(bool enable_fim) {
     llh = getNaN();
     chi2 = getNaN();
     if (sensi >= SensitivityOrder::first) {
         sllh.resize(nplist, getNaN());
         if (sensi >= SensitivityOrder::second)
             s2llh.resize(nplist * (nJ - 1), getNaN());
 
-        if (sensi_meth == SensitivityMethod::forward ||
-            sensi >= SensitivityOrder::second)
+        if ((sensi_meth == SensitivityMethod::forward ||
+            sensi >= SensitivityOrder::second) & enable_fim)
```

Comment:
```suggestion
            sensi >= SensitivityOrder::second) && enable_fim)
```

Should work either way, but let's stick to logical operators.